### ISSUE TITLE
OGC polygon filter not used with Sql Server

### ISSUE THREAD

**Issue by PhilAnzel** [mapserver/mapserver#4603]
In MapServer 6.0.3 (in MS4W) and in MapServer 6.2 it appears that a WFS FILTER polygonal extent is not used in SQL Server 2008 data retrieval.I believe it had worked properly under MapServer 5.6.0.

For example, consider a mapfile with overall projection 4326 (i.e., lat/long), an "EXTENT -180.0 20.0 -50.0 60.0", and all data layers coming from SQL Server with SRS=4326.

In the case of a WFS version 1.0 request with a polygonal FILTER=
```
  <Filter>
    <Intersect>
       <PropertyName>Geometry</PropertyName>
        <gml:Polygon>
          <gml:outerBoundaryIs>
            <gml:LinearRing>
              <gml:coordinates>
                 -88.0865046533,37.5555143852
                 +-88.0860204771,37.5600435404
                 +-88.0782858287,37.5595392364
                 +-88.0787704736,37.5550101113
                 +-88.0865046533,37.5555143852
              </gml:coordinates>
            </gml:LinearRing>
          </gml:outerBoundaryIs>
        </gml:Polygon>
      </Intersect>
    </Filter>
```
the SQL statement sent to the DATA source in a SQL Server database uses the mapfile's EXTENT when retrieving data:
  ```WHERE <spatialColumn>.STIntersects(geometry::STGeomFromText('POLYGON((-180 20,-50 20,-50 60,-180 60,-180 20))',4326)) = 1```

This is a significant performance problem, especially if the extent roughly covers the entire globe and the polygon of interest is a small one.

By comparison, if the Filter=
```
  <Filter>
    <BBOX>
      <PropertyName>Geometry</PropertyName>
      <Box srsName='EPSG:4326'>
        <coordinates>-86.46878,40.56448 -86.45282,40.57646</coordinates>
      </Box>
    </BBOX>
  </Filter>
```
the emitted SQL statement uses the exact filter's request:
```('POLYGON((-86.46878 40.56448,-86.45282 40.56448,-86.45282 40.57646,-86.46878 40.57646,-86.46878 40.56448))',4326)) = 1 ```

**Comment by sdlime** [comment]
The RFC 91 work for MapServer 7.0 provides the opportunity to fix this. There isn't a MS SQL Server transformation function yet but it could be added (PostGIS and Oracle Spatial provide a pattern) relatively easily I think. I'm not familiar enough with SQL Server spatial support to do that work though.

Steve

**Comment by PhilAnzel** [comment]
I can possibly help with the Microsoft "Transact-SQL" ("T-SQL") queries against SQL Server spatial data. SQL Server professes implementation of a full set of OGC methods, so I could give it a go. 

When I first submitted this bug I thought that all that might be needed is an filter expressed as WKT, then the query against the SQL Server database could use the pair of statements:
> declare @filter geometry  = geometry::STGeomFromText(@wkt, <srid>);
> select <column list> from <table> where <spatial column>.STIntersects(@filter) = 1;

Keep me posted, I'm still eager to get this issue resolved as the current non-BBOX FILTER is extermely expensive when performed against tables with tens of millions of polygons with some polygons reaching hundreds of thousands of vertices.

**Comment by szekerest** [comment]
I think this has already been implemented as per #4974 
Closing
